src/rules/no-import-module-exports.js

```diff
@@ -0,0 +1,72 @@
+import minimatch from 'minimatch'
+import path from 'path'
+import readPkgUp from 'read-pkg-up'
+
+function getEntryPoint(context) {
+  try {
```

Comment:
pkg-up is already a dep of read-pkg-up, and the cost of adding dependencies to a dev dependency is virtually zero. I'd say C.

However, I just realized my comment on the original issue does also add that it should probably support `jsnext:main`, and for that, you'd need A or B.

I'd still lean towards C for the primary case.